keps/sig-node/20190604-cgroup-none.md

```diff
@@ -0,0 +1,174 @@
+---
+title: New cgroup driver: "none" (for Rootless)
+authors:
+  - "@AkihiroSuda"
+owning-sig: sig-node
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-06-04
+last-updated: 2019-06-04
+status: implementable
+---
+
+# New cgroup driver: "none" (for Rootless)
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Future: cgroup2](#future-cgroup2)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details](#implementation-details)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+- [Feature Gate](#feature-gate)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Summary
+
+Add "none" cgroup driver to kubelet so that kubelet (as well as CRI/OCI runtimes) can be executed without write permissions to cgroups.
+
+Docker [v19.03+](https://github.com/moby/moby/pull/38050), containerd/CRI [20190104+](https://github.com/containerd/cri/pull/970), and CRI-O [v1.12+](https://github.com/cri-o/cri-o/pull/1729) already support running without cgroup permissions. So we only need to update kubelet.
+
+Draft patch for kubelet is available in [Usernetes project](https://github.com/rootless-containers/usernetes): https://github.com/rootless-containers/usernetes/blob/2fd0ba0594673e0c5ea04a894191bfaa3b117377/src/patches/kubernetes/0003-kubelet-new-feature-gate-SupportNoneCgroupDriver.patch
+
+An old version of our patch has been also used by Rancher's [k3s](https://github.com/rancher/k3s/commit/e397cb4e5b017805588ff223a63f42bff1f42d3a).
+
+## Motivation
+
+The motivation of adding "none" cgroup driver is to allow running the entire Kubernetes components (`kubelet`, CRI, OCI, CNI, and all `kube-*`) as a non-root user (w/o write access to `/sys/fs/cgroup`) on the host. ("Rootless mode").
+
+The motivations of Rootless mode are:
+* To protect the host from potential container-breakout vulnerabilities
+* To allow users of shared machines (especially HPC) to run Kubernetes without the risk of breaking their colleagues' environments
+* Safe Kubernetes-on-Kubernetes
+
+Without cgroups, kubelet cannot support restricting and monitoring CPU/memory resources for pods, but pod processes can still spontaneously impose traditional `setrlimit` restrictions on themselves.
```

Comment:
Is a "rootless" kubernetes cluster with cgroups disabled intended to pass any node e2e or cluster e2e conformance tests?